classes/ImageManager.php

```diff
@@ -451,7 +451,7 @@ public static function validateUpload($file, $maxFileSize = 0, $types = null)
             return Context::getContext()->getTranslator()->trans('Image is too large (%1$d kB). Maximum allowed: %2$d kB', array($file['size'] / 1024, $maxFileSize / 1024), 'Admin.Notifications.Error');
         }
         if (!ImageManager::isRealImage($file['tmp_name'], $file['type']) || !ImageManager::isCorrectImageFileExt($file['name'], $types) || preg_match('/\%00/', $file['name'])) {
-            return Context::getContext()->getTranslator()->trans('Image format not recognized, allowed formats are: .gif, .jpg, .png', array(), 'Admin.Notifications.Error');
+            return Context::getContext()->getTranslator()->trans('Image format not recognized, allowed format(s) is(are): .%s', array(implode(', .', $types)), 'Admin.Notifications.Error');
```

Comment:
Unfortunately, when `$types` is null, this will have weird consequences on the message displayed. This PR requires more changes to at least get the default list in that case.

Reply:
Then, it's maybe better to move this part  
` if ($authorizedExtensions === null) {
    $authorizedExtensions = array('gif', 'jpg', 'jpeg', 'jpe', 'png');
}`
 from the method `isCorrectImageFileExt` of the same class to this method 
`validateUpload` 
like 
`if ($types === null) {
            $types = array('gif', 'jpg', 'jpeg', 'jpe', 'png');
        }` and in this case, we will have default file extensions list before it's verification. What do you think?

Reply:
Indeed, we could list these values in a class constant, that we could then reuse easily in the different methods of the class.

```php
class ImageManagerCore
{
    const DEFAULT_ALLOWED_EXTENSIONS = [gif', 'jpg', 'jpeg', 'jpe', 'png'];

    [...]
```

Then defining the list as you suggested with

```php
if ($authorizedExtensions === null) {
    $authorizedExtensions = self::DEFAULT_ALLOWED_EXTENSIONS;
}
```